service/controller/v14patch2/key/key.go

```diff
@@ -192,6 +192,20 @@ func IsPodDrained(pod *corev1.Pod) (bool, error) {
 		return false, microerror.Mask(err)
 	}
 
+	// As a safety net check POD's container statuses. If all containers are
+	// Terminated, POD can be deleted even if annotation is missing still.
```

Comment:
This is not true because above we return in case the annotation is missing. I am also thinking to have a separate method for this like `IsPodTerminated` or `IsPodDead` or something like this. We should not mix semantics here and rather have a dedicated check which we leverage where we need it.

Reply:
Oh, this was a good catch. I'm also fine with putting this to separate function. Will start fixing this. See you soon :heart: